File: src/main/golang/leetcode/P496_Next_Greater_Element_I.go
package leetcode

func nextGreaterElement(nums1 []int, nums2 []int) []int {
	result := make([]int, len(nums1))

	for i, len, len2 := 0, len(nums1), len(nums2); i < len; i++ {
		flag := false

		for j := 0; j < len2; j++ {
			if flag && nums1[i] < nums2[j] {
				result[i] = nums2[j]
				break
			}

			if nums1[i] == nums2[j] {
				flag = true
			}
		}

		if result[i] == 0 {
			result[i] = -1
		}
	}

	return result
}

File: src/main/javascript/leetcode/P496_Next_Greater_Element_I.js
var nextGreaterElement = function (nums1, nums2) {
    let result = [];

    for (let i = 0, len = nums1.length; i < len; i++) {
        let flag = false;

        for (let j = 0, len2 = nums2.length; j < len2; j++) {
            if (flag && nums1[i] < nums2[j]) {
                result[i] = nums2[j];
                break;
            }

            if (nums1[i] === nums2[j]) {
                flag = true;
            }
        }

        if (result[i] === undefined) {
            result[i] = -1;
        }
    }

    return result;
};

console.log(nextGreaterElement([4, 1, 2], [1, 3, 4, 2]));